c/src/point_one/polaris/polaris.c

```diff
@@ -543,26 +554,40 @@ int Polaris_Work(PolarisContext_t* context) {
 #endif
 
   if (bytes_read < 0) {
+    // Did we hit a read timeout? This is normal behavior (e.g., brief loss of
+    // cell service) and is not considered an error. Most receivers can handle
+    // small gaps in corrections data. We do not close the socket here.
+#ifdef POLARIS_USE_TLS
+    int ssl_error = SSL_get_error(context->ssl, bytes_read);
+    if (ssl_error == SSL_ERROR_WANT_READ || ssl_error == SSL_ERROR_WANT_WRITE) {
```

Comment:
Are you sure that when USE_TLS we don't also need to be checking errno? If so we could consider checking if errno != EAGAIN/TIMEOUT first, and if that isnt the case, then checking the ssl_error states.

Reply:
Yes -- `SSL_read()` does not set errno for timeouts. It may set errno for other conditions, in which case `bytes_read` will be <0 (but SSL does not guarantee what value it will take beyond that) and the returned `ssl_error` value will be `SSL_ERROR_SYSCALL`. We treat that like any other non-timeout error condition.

Reply:
ok that works for me. As long as the ssl errors call traps underlying socket issues then no problem as is.